src/Control/Bind.purs

```diff
@@ -63,6 +63,17 @@ infixr 1 bindFlipped as =<<
 instance bindFn :: Bind ((->) r) where
   bind m f x = f (m x) x
 
+-- | The `Array` monad works like a nested for loop. Each
+-- | `bind`/`>>=` adds another level of nesting in the loop:
+-- | ```
+-- | foo :: Array Int
+-- | foo = do
+-- |   ["a", "b"] >>= \eachElementInArray1 ->
+-- |     ["c", "d"] >>= \eachElementInArray2
+-- |       pure (eachElementInArray1 <> eachElementInArray2)
+-- |
+-- | foo == [ ("a" <> "c"), ("a" <> "d"), ("b" <> "c"), ("b" <> "d") == [ "ac", "ad", "bc", "bd"]
```

Comment:
I think at this point the `[ ("a" <> "c"), ... ]` is probably just adding noise - part of the reason I suggested changing the operation in this example is so that it could be done away with and we could just write `foo == [ "ac", "ad", "bc", "bd" ]`. Also, I think it would be good to have the spaces either side of the [] match - at the moment there is a space after the [ but there is not a space before the ].

Reply:
I decided to do this:
```
-- | -- In other words...
-- | foo
-- | -- ... is the same as...
-- | [ ("a" <> "c"), ("a" <> "d"), ("b" <> "c"), ("b" <> "d") ]
-- | -- which simplifies to...
-- | [ "ac", "ad", "bc", "bd" ]
```